osu.Game/Screens/Select/CarouselContainer.cs

```diff
@@ -184,6 +185,58 @@ public void SelectGroup(BeatmapGroup group, BeatmapPanel panel, bool animated =
             ScrollTo(selectedY, animated);
         }
 
+        public void Sort(FilterControl.SortMode mode) {
+            switch (mode) { 
+                case FilterControl.SortMode.Artist:
+                    groups.Sort((x, y) =>
+                    {
+                        return string.Compare(x.BeatmapSet.Metadata.Artist, y.BeatmapSet.Metadata.Artist);
+                    });
+                    break;
+                case FilterControl.SortMode.Title:
+                    groups.Sort((x, y) =>
+                    {
+                        return string.Compare(x.BeatmapSet.Metadata.Title, y.BeatmapSet.Metadata.Title);
+                    });
+                    break;
+                case FilterControl.SortMode.Author:
+                    groups.Sort((x, y) =>
+                    {
+                        return string.Compare(x.BeatmapSet.Metadata.Author, y.BeatmapSet.Metadata.Author);
+                    });
+                    break;
+                case FilterControl.SortMode.Difficulty:
+                    groups.Sort((x, y) =>
+                    {
+                        if (x.BeatmapSet.Beatmaps.First().BaseDifficulty.OverallDifficulty >

```

Comment:
It's probably a good idea to leave a TODO comment there so we can replace `OverallDifficulty` by the actual star rating whenever it gets implemented.